firedrake/slate/preconditioners.py

```diff
@@ -389,10 +356,14 @@ def apply(self, pc, x, y):
             broken_pressure.dat.copy(unbroken_pressure.dat)
 
             # Compute the hdiv projection of the broken hdiv solution
-            self._assemble_projection_rhs()
-            with self._projection_rhs.dat.vec_ro as b_proj:
-                with self.unbroken_solution.split()[self.vidx].dat.vec_wo as sol:
-                    self.hdiv_projection_ksp.solve(b_proj, sol)
+            broken_hdiv = self.broken_solution.split()[self.vidx]
+            unbroken_hdiv = self.unbroken_solution.split()[self.vidx]
+            unbroken_hdiv.assign(0)
+
+            par_loop(self.average_kernel, ufl.dx,
+                     {"w": (self.weight, READ),
+                      "vec_in": (broken_hdiv, READ),
+                      "vec_out": (unbroken_hdiv, INC)})
```

Comment:
Oh I see, you use it here too.  In this case, you do need to zero and increment, rather than just assign.